### ISSUE TITLE
Allow controlled resizing

### ISSUE THREAD

**Issue by yang** [bokuweb/re-resizable#319]
Currently re-resizable exclusively owns the width/height during a resize.  It would be powerful to make this controllable, so that users can implement (for instance) their own sizing (snapping, clamping, etc.) logic.

(As a hack for now, one can override the `calculateNewSize` method to always return `auto`, and then explicitly control the size of a grand-child div from the `onResize` callback.)

**Comment by jakub-chatrny** [comment]
One usecase where you can use it is when you have absolute position of box and you are resizing (left, top). You might want change (left, top) position.

**Comment by mdanka** [comment]
This is also a problem when using a scale transform.

For example, with a scale of 2:
![reresizable](https://user-images.githubusercontent.com/8069999/43006251-8092451a-8c35-11e8-957b-ffeb49a9ca55.gif)

**Comment by jakub-chatrny** [comment]
Yep I face same issue using react-rnd (which extends re-resizable)
Thanks for providing workaround @mdanka !

**Comment by chathuraa** [comment]
Another use case is when implementing auto scroll functionality to sync resizing with scroll speed.

**Comment by JoshuaCWebDeveloper** [comment]
It wasn't immediately clear to me from reading this ticket, so I just wanted to clarify that this functionality does appear to be currently implemented in the latest version (using the `size` property and the resize handlers).

**Comment by codebycarlos** [comment]
Agree with @JoshuaCWebDeveloper this can be accomplished by using the size property and setting this with your own state (i.e. width, height) then on the onResize event you can change dimensions as desired.

**Comment by vincaslt** [comment]
It isn't really contolled, the problem is that it maintains internal state and does the sizing based on that. It does accept and synchronize outside state via props, but while resizing it's ignored and the internal state is used, meaning that I'm not really controlling the size, for instance I can't stop resizing at some point (e.g. reached some boundary), because the internal state keeps going.
